integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java

```diff
@@ -381,6 +383,19 @@ private void initializeContextFactoryIfNotInitialized(SparkConf sparkConf, Strin
   private static void initializeMetrics(OpenLineageConfig<?> openLineageConfig) {
     meterRegistry =
         MicrometerProvider.addMeterRegistryFromConfig(openLineageConfig.getMetricsConfig());
+
+    // register SimpleMeterRegistry if no other registries are present and debug facet is enabled
+    if (meterRegistry instanceof CompositeMeterRegistry
+        && ((CompositeMeterRegistry) meterRegistry).getRegistries().isEmpty()
+        && openLineageConfig.getFacetsConfig() != null
+        && openLineageConfig.getFacetsConfig().getDisabledFacets() != null
+        && openLineageConfig.getFacetsConfig().getDisabledFacets().containsKey("debug")
+        && !openLineageConfig.getFacetsConfig().getDisabledFacets().get("debug")) {
+      meterRegistry =
+          new CompositeMeterRegistry(
+              Clock.SYSTEM, Collections.singletonList(new SimpleMeterRegistry()));
```

Comment:
Please don't overwrite `meterRegistry` - it won't be visible for anyone else who uses `MicrometerProvider.getMeterRegistry`. You can simply add that `SimpleMeterRegistry` using `MicrometerProvider.addMeterRegistry`.